### DISCUSSION TITLE
Check if a Recoil value is available without triggering it?

### DISCUSSION THREAD

**Discussion by phuctm97** [facebookexperimental/Recoil#2182] [Q&A]
Is there anyway to check if a Recoil value (atom/selector) is available without triggering its effects/default value in a `selector`?

Currently in selector there is only `get` method will always trigger the atom effects/default value, would be great to have a `has` or `isSet`/`isDefault` getter that only checks & subscribers to a value without triggering it.

Current workaround is to create a separate boolean atom just to start that value and update when it's appropriate, but it feels redundant.

**Comment by tpatalas** [comment]
I would use atom without default value and return undefined if there is no value set.
